uint8 token_kobuki
uint8 counter